maggy/ablation/ablationstudy.py

```diff
@@ -203,9 +203,28 @@ def __init__(self):
         self.layers = Layers()
         self.base_model_generator = None
 
+        # the list holding custom model generators. 
+        # One (extra) trial will be generated per each item in the list.
+        self.custom_model_generators = [] 
+
     def set_base_model_generator(self, base_model_generator):
         self.base_model_generator = base_model_generator
-
+    
+    def add_custom_model_generator(self, custom_model_generator, model_identifier):
+        """
+        Add a custom model architecture generator, which will result in a single ablation trial. 
+        The `custom_model_generator` should be a Python callable, that once called should return a 
+        `keras.Model` object. This provides support for ablation study of 
+        arbitrarily complex models developed by extending `tf.Model` (for an example see 
+        [Writing custom layers and models with Keras](https://www.tensorflow.org/guide/keras/custom_layers_and_models#building_models)).
+        *Side effect*: This callable will be appended to `ablationstudy.Model.custom_model_generators`.
```

Comment:
The user doesn't need to know about the internals. Desribe more in general that each custom generator will be a separate trial in the study.